src/App.tsx

```diff
@@ -9,52 +13,102 @@
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
 
-export const App = () => (
-  <main className="section">
-    <div className="container">
-      <div className="tile is-ancestor">
-        <div className="tile is-parent">
-          <div className="tile is-child box is-success">
-            <div className="block">
-              <UserSelector />
-            </div>
+export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [currUser, setCurrUser] = useState<User | null>(null);
+  const [currPost, setCurrPost] = useState<Post | null>(null);
+  const [openModal, setOpenModal] = useState(false);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState(false);
 
-            <div className="block" data-cy="MainContent">
-              <p data-cy="NoSelectedUser">No user selected</p>
+  const searchUserPosts = (userId: number) => {
+    getUserPosts(userId)
+      .then(setPosts)
+      .catch(() => setError(true))
+      .finally(() => setLoading(false));
+  };
 
-              <Loader />
+  useEffect(() => {
+    getUsers().then(setUsers);
```

Comment:
The `getUsers` function lacks error handling. It's important to handle errors here to provide feedback to users in case of failures. Consider adding a `.catch()` block to handle potential errors.